### DISCUSSION TITLE
ipam/aggregates  by prefix values

### DISCUSSION THREAD

**Discussion by vladimirs-git** [netbox-community/netbox#14305] [Help Wanted!]
I am trying to retrieve multiple aggregates by prefix values: "100.64.0.0/10", "10.0.0.0/8", but Netbox only returns the last prefix "10.0.0.0/8".
https://demo.netbox.dev/ipam/aggregates/?prefix=100.64.0.0%2F10&prefix=10.0.0.0%2F8

For prefixes and addresses, the same approach is working as expected. Netbox returns all requested prefixes.
https://demo.netbox.dev/ipam/prefixes/?prefix=10.112.160.0%2F28&prefix=10.112.142.0%2F24
https://demo.netbox.dev/ipam/ip-addresses/?address=1.2.3.4%2F30&address=1.2.3.4%2F29

Could you please clarify why aggregates behave differently?

**Comment by candlerb** [comment]
These work for me when used separately:

* https://demo.netbox.dev/ipam/aggregates/?prefix=100.64.0.0%2f10
* https://demo.netbox.dev/ipam/aggregates/?prefix=10.0.0.0%2f8

So it seems that this particular filter doesn't work as an "OR" when given multiple values. You could raise this as a feature request (or bug).

I note that there are some other things which could add to the confusion.

Firstly, aggregates and prefixes are two different object types, and in the demo instance there is an aggregate 100.64.0.0/10 but no prefix 100.64.0.0/10. See https://demo.netbox.dev/ipam/aggregates/4/prefixes/  (But actually this doesn't affect the issue under consideration)

Secondly, "prefix" is also an attribute of Aggregate, and is actually the identity of the aggregate (although it is non-unique). But in the table view, this is labelled as "Aggregate" rather than "Prefix"!
